enhancements/baremetal/baremetal-provisioning-config.md

```diff
@@ -0,0 +1,220 @@
+within the operator scope and will only contain configuration required by the
+provisioning service to boot baremetal servers.
+
+### Non-Goals
+
+The provisioning network or the provisioning IP are not expected to change after
+deployment. But, it is possible for the DHCP range to be expanded after nodes
+have already been deployed. This proposal is not considering the update path for
+these configurations items.
+
+### Proposal
+
+Before BareMetal Hosts can be matched to Machines, they need to be connected to their
+provisioning network for them to be PXE booted and given an IP address. To make
+this happen, the provisioning service needs to know which NIC, provisioning network,
+IP address and image URLs to use to download and boot images on these servers.
+
+A new CR called "Metal3ProvisioningController" would be created within the operator scope.
+
+This new CR would consist of the following:
+
+1. ProvisioningInterface : This is the interface name on the underlying baremetal
+server which would be physically connected to the provisioning network. This
+configuration is needed only for the underlying provisioning service (Ironic)
+and could have values like "eth1" or "ens3".
+
+2. ProvisioningIP : This is the IP address used to to bring up a NIC on the
+baremetal server for provisioning. This is also a value that is useful just to the
+provisioning service. This value should not be in the DHCP range and should not
+be used in the provisioning network for any other purpose (should be a free IP
+address.) It is expected to be provided in the format : 10.1.0.3.
+
+3. ProvisioningNetworkCIDR : This is the provisioning network with its CIDR. The
+ProvisioningIP and the ProvisioningDHCPRange are expected to be within this network.
+The value for this config is expected to be provided in the format : 10.1.0.0/24.
+
+4. ProvisioningDHCPExternal : When this is set to True, no DHCP services would be
+started within the metal3 cluster. Ironic services will depend on an External DHCP
+server to provide the IP addresses to be assigned to baremetal hosts. When the value
+of this flag is True, the value provided in ProvisioningDHCPRange would be ignored.
+The value of this configuration is not expected to change after the install process.
+
+5. ProvisioningDHCPRange : This is a range of IP addresses from which baremetal
+hosts can be assigned their IP address. This range should be provided as two
+comma seperated IP addresses where the addresses represent a contiguous range of
+IPs that are available to be allocated. It is an optional parameter and it could
+have the following values:
+	1. No DHCP in cluster (ProvisioningDHCPExternal is True), its value will be
+	ignored.
+	2. DHCP within the cluster (ProvisioningDHCPExternal is False), and its value
+	is unset. Default DHCP range (.10 to .100) would be in use.
+	3. DHCP within the cluster (ProvisioningDHCPExternal is False), and a specific
+	DHCP range is specified. For example: 10.1.0.10, 10.1.0.100.
```

Comment:
what about something like this:
```yaml
provisioningDHCP:
    type: Removed
```

```
provisioningDHCP:
    type: Managed
    range: <something>
```

the goal is to mimic states like https://github.com/openshift/api/blob/992bc3a41fe6c4e5bcf27dc8080f9e5c114c33fc/operator/v1/types.go#L30-L44 where the user can unmanage,remove, manage certain operator managed services???

Reply:
Yes, that seems like a good way to do it. Can we add our own management state "External" to https://github.com/openshift/api/blob/992bc3a41fe6c4e5bcf27dc8080f9e5c114c33fc/operator/v1/types.go#L30-L44. If not we might use "Unmanaged" instead of "Removed".

Reply:
from a point of management, Removed has stronger/clearer meaning ie `I have my own DHCP and please remove any of your managed solution.`  whereas External implies the same thing...

> Can we add our own management state "External"

that's is probably going to be tough..

Reply:
Ok. Using the existing Management type supported values. We are sticking with "Removed" instead of "Unmanaged".

Reply:
I do like using an existing API but reading the docs, it sounds like if we were to follow it exactly, we'd have to support Force and Managed, as neither has a comment saying it's optional. Force doesn't make a lot of sense to us I guess.  Removed is also not super clear to me.

Especially since this is going into the platform installconfig, I'd expect users to be interacting with it directly as part of initial installation, I feel like a single boolean conveys the information much more simply.

Reply:
I am also OK with the idea of adding our own "DHCP type" having full control over the keywords that are allowed.

Reply:
I think @abhinavdahiya's approach is fine, and I see you already updated the doc. Let's just go with that, I just left a small comment there about including an example.  Can you mark the DHCP conversation threads in this PR resolved?